lib/Star/Kata/Builder/KataBuilder.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * This file is part of the phpkata project.
+ *
+ * (c) Yannick Voyer (http://github.com/yvoyer)
+ */
+
+namespace Star\Kata\Builder;
+
+use Star\Kata\Model\Kata;
+use Star\Kata\Model\Objective\Objective;
+
+/**
+ * Class KataBuilder
+ *
+ * @author  Yannick Voyer (http://github.com/yvoyer)
+ *
+ * @package Star\Kata\Builder
+ */
+final class KataBuilder
+{
+    /**
+     * @var ObjectiveBuilder
+     */
+    private $objectiveBuilder;
+
+    /**
+     * @var array
+     */
+    private $data;
+
+    /**
+     * @var Objective[]
+     */
+    private $objectives = array();
+
+    /**
+     * @var int
+     */
+    private $count = 1;
+
+    public function __construct()
+    {
+        $this->resetData();
+        $this->objectiveBuilder = new ObjectiveBuilder();
+    }
+
+    /**
+     * @return Kata
+     */
+    public function build()
+    {
+        $this->count ++;
+        // todo define path in builder construct?
+        $kata = new Kata('path', $this->data['name']);
+        foreach ($this->objectives as $objective) {
+            $kata->addObjective($objective);
+        }
+
+        $this->resetData();
```

Comment:
But keeping references of the current configuration, might add a unexpected behavior. What if i were to build 2 kata with the same objective. There would be only one objective referenced in two kata. If the objective of kata 1 was to pass, i don't want the objective of kata 2 to pass, but since its the same reference, they both would.